### PR TITLE
harness.setNow() mocks global time via sinon.useFakeTimers

### PR DESCRIPTION
**Pull request by kennsippell**
This is a proposed breaking change to the conf harness that changes "setNow" from mocking just "certain functions" to a global mocking of DateTime.

#12 

Currently,  `harness.setNow('2000-01-01')` would mock the result returned by `Utils.now()` and mocks the time available within enketo forms, but it does not mock the global `new DateTime()`.  This is particularly confusing when some code uses Utils.now(), but other does not (like contact summary which does not have that library available). 

```
      const mockedTime = 1000;
      await harness.setNow(mockedTime);
      expect(new Date().getTime()).to.eq(mockedTime);
      await harness.clear();
      expect(new Date().getTime()).to.not.eq(mockedTime);
```

I believe the global mocking of time is a more consistent and intuitive experience, but this is a breaking change to the harness.

Published as `medic-conf-test-harness@1.0.0-beta.0` for experimentation.

### CODE REVIEW

File: src/harness.js [hunk 1]
```diff
@@ -187,13 +194,20 @@ class Harness {
...
    */
   setNow(now) {
+    if (!now) {
+      throw Error('undefined date passed to setNow');
+    }
+
     const parseableNow = typeof now === 'object' ? now.getTime() : now;
-    this._now = toDate(parseableNow).getTime();
-    return this.page.evaluate(innerNow => window.now = new Date(innerNow), this._now);
+    const asTimestamp = toDate(parseableNow).getTime();
```
Comment (eliaswalyba) [inline src/harness.js:207]: Not really relevant but couldn't we avoid `asTimestamp` variable here and just pass `toDate(parseableNow).getTime()` directly to `this._now` ?

**Review by eliaswalyba — APPROVED** [review]

**Comment by eliaswalyba** [comment]
This is very fine to me. Very clean and concise!

File: src/harness.js (hunk 1 above)
Comment (eljhkrr) [reply src/harness.js:207]: Wouldn't this mean redoing the computation again in L209? Variables help by storing the result for reuse

**Review by eljhkrr — APPROVED** [review]
Looks good

File: src/harness.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const { getPatientId } = require('../ext/registration-utils');
 const getNoolsInstances = require('./get-nools-instances');
 const mergeInstanceToTarget = require('./merge-instances-to-target');
 const toDate = require('./toDate');
+const sinon = require('sinon');
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/harness.js:11]: hmmm... interesting suggestion. looks like it doesn't have a singleton-pattern uninstall...

File: src/harness.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/harness.js:207]: Is this simpler?

**Pull request merged**